Source/Particles/RigidInjectedParticleContainer.cpp

```diff
@@ -233,66 +236,91 @@ RigidInjectedParticleContainer::PushPX(WarpXParIter& pti,
     Cuda::ManagedDeviceVector<Real> xp_save, yp_save, zp_save;
     RealVector uxp_save, uyp_save, uzp_save;
 
+    Real* AMREX_RESTRICT x = xp.dataPtr();
+    Real* AMREX_RESTRICT y = yp.dataPtr();
+    Real* AMREX_RESTRICT z = zp.dataPtr();
+    Real* AMREX_RESTRICT gi = giv.dataPtr();
+    Real* AMREX_RESTRICT uxpp = uxp.dataPtr();
+    Real* AMREX_RESTRICT uypp = uyp.dataPtr();
+    Real* AMREX_RESTRICT uzpp = uzp.dataPtr();
+    Real* AMREX_RESTRICT Expp = Exp.dataPtr();
+    Real* AMREX_RESTRICT Eypp = Eyp.dataPtr();
+    Real* AMREX_RESTRICT Ezpp = Ezp.dataPtr();
+    Real* AMREX_RESTRICT Bxpp = Bxp.dataPtr();
+    Real* AMREX_RESTRICT Bypp = Byp.dataPtr();
+    Real* AMREX_RESTRICT Bzpp = Bzp.dataPtr();
```

Comment:
It looks like `Exp` in this function is only used once, to define 
```
Real* AMREX_RESTRICT Expp = Exp.dataPtr();
```
Could we define `Exp` as 
```
Real* AMREX_RESTRICT Exp = attribs[PIdx::Ex].dataPtr();
```
and use it everywhere, and get rid of `Expp` for clarity? 
Same question for `Eyp` `Ezp` `Bxp` `Byp` `Bzp`.

Reply:
I've made the changes, was well as additional ones for clarity and optimization.